src/server/scripts/Kalimdor/boss_azuregos.cpp

```diff
@@ -62,16 +62,13 @@ class boss_azuregos : public CreatureScript
 
         void Reset() override
         {
-            me->RemoveAurasDueToSpell(SPELL_MARK_OF_FROST_AURA);
-            _scheduler.CancelAll();
```

Comment:
Do not remove this.

Reply:
Can you give me some context why?  Unless I am missing something, that's what is causing the non blizzlike behavior.
I did a lot of testing with raids and parties and I am certain this is what's clearing it on wipe.  Even now during the anniversary events the debuff sticks for 15 minutes through death.

I am still learning your ways but I am a bit confused.

Reply:
Okay, I only asked for the change of line 66, not the one above. Basically _scheduler.CancelAll() just remove all the scheduled tasks the boss should do (casting spells). Since we are resetting it, we need to cancel all his spell casts and so. If we don't do this, it might bug next time we engage fight (doubtful now, since the scheduler schedules (yikes) the tasks again when you enter combat with him. Don't remove _scheduler.CancelAll() or it might cause undesirable bugs.

Reply:
Understood.  That makes total sense...I was under the impression the CancelAll was purely related to the player auras and debuffs...

Reply:
CancelAll restored.  Ready for Testing.